blog/posts/swiip/the-weight-of-js.md

```diff
@@ -0,0 +1,191 @@
+
+They have almost no network consideration to deal with (as far as I know) yet they care deeply at their app size for performances and fluidity.
+
+Lastly: "less code is less bug". I always loved this quotation. Shipping less code (both yours and third parties) will always be a way to a better stability.
+
+## Tools don't solve everything
+
+A good part of the so rich Web tooling we use nowadays is about limiting the size of your JS. All of these techniques are optimizations which is great and I recommend to use almost all of them. Yet, optimizations are not solutions and the real amount of JS is still a question.
+
+### Bundling
+
+The first tool we use about JS performance is bundling. We don't serve the browser as many files as we develop knowing that every request to the server has its cost. Most of the time, we compile a bundle of one, two to five bundled JS file. Yet, you still has as much code as before.
+
+### Minifying
+
+Also for a long time, we use JS minification. It means to modify the code by removing most of the spaces, line breaks, rename variables to shorter names to have the same code with less characters and then, a smaller weight. It's great, it has almost no counterpart unless readability and then debuggability but it can be handled by the use of source maps.
+
+Minification will reduce the weight of your JS from 50% to 90%! (if you have lots of comments 😀)
+
+<iframe width="800" height="450" src="//embed.chartblocks.com/1.0/?c=5ce570fc3ba0f6d07e63d2c9&t=a95c1edce5b7d21" frameBorder="0"></iframe>
+
+### Compression
+
+The most magical solution to reduce JS weight is binary compression. Most of the time, you don't even have to configure any tooling, it's your HTTP server which will compress your files with Gzip or Brotli. For best practices about compression, see this great article of [Antoine Caron](https://open-blog.dev/authors/slashgear) [Why you should use Compression Webpack Plugin?](https://open-blog.dev/posts/slashgear/why-you-should-use-compression-webpack-plugin/).
+
+Like others, compression is great, you should definitely use it. Nevertheless, it doesn't mean you have free hand about the amount of JS you serve.
+
+Firstly because no matters how great the percentage of reduction the compression offer, it's still a ratio of your original weight: 30% of 10MB is still 3MB…
+
+<iframe width="800" height="450" src="//embed.chartblocks.com/1.0/?c=5ced4a1f3ba0f66a7e57560b&t=95e776621bbc403" frameBorder="0"></iframe>
+
+Secondly because compression is only a network artifact. The amount of data to transfert on the wire is reduced but your actual amount of code is still the same. The browser will still have to parse and compile the same amount of code. Worst, the client will have to decompress the file before being able to use it. These steps could seem unimportant but on an old smartphone, it can be really slow, sometime longer than network time!
```

Comment:
```suggestion
Secondly because compression is only a network artifact. The amount of data to transfert on the wire is reduced but your actual amount of code is still the same. The browser will still have to parse and compile the same amount of code. Even worse, the client will have to decompress the file before being able to use it. These steps could seem unimportant but on an old smartphone, it can be really slow, sometime longer than network time!
```